bokehjs/src/coffee/models/tools/actions/undo_tool.ts

```diff
@@ -0,0 +1,28 @@
+import {ActionTool, ActionToolView} from "./action_tool"
+
+export class UndoToolView extends ActionToolView {
+
+  model: UndoTool
+
+  connect_signals(): void {
+    super.connect_signals()
+    this.connect(this.plot_view.state_changed, () => this.model.disabled = !this.plot_view.can_undo())
+  }
+
+  doit(): void {
+    this.plot_view.undo()
+  }
+}
+
+export class UndoTool extends ActionTool {
+  tool_name = "Undo"
+  icon = "bk-tool-icon-undo"
+}
+
+UndoTool.prototype.type = "UndoTool"
+
+UndoTool.prototype.default_view = UndoToolView
+
+UndoTool.override({
+  disabled: true,
+})
```

Comment:
FWIW undo/redo don't seem to be working for me on `color_scatter.py`

Reply:
Yeah, it kind of works, but also doesn't (maybe one step works).

Reply:
I will look into this, #7355.